src/algebra/order/sub.lean

```diff
@@ -557,6 +557,10 @@ contravariant.add_le_cancellable.tsub_tsub_assoc h₁ h₂
 lemma tsub_add_tsub_comm (hba : b ≤ a) (hdc : d ≤ c) : a - b + (c - d) = a + c - (b + d) :=
 contravariant.add_le_cancellable.tsub_add_tsub_comm contravariant.add_le_cancellable hba hdc
 
+protected lemma tsub_tsub_tsub_cancel_left (hba : b ≤ a) (hcb : c ≤ b) : a - c - (a - b) = b - c :=
+by rw [tsub_eq_iff_eq_add_of_le (tsub_le_tsub_left hcb a), tsub_add_eq_add_tsub hcb, add_comm,
+  tsub_add_cancel_of_le hba]
```

Comment:
Does this work?
```suggestion
lemma tsub_tsub_tsub_cancel_left (hba : b ≤ a) : a - c - (a - b) = b - c :=
by rw [←tsub_add_eq_tsub_tsub, add_comm, tsub_add_eq_tsub_tsub, tsub_tsub_cancel_of_le hba]
```

Reply:
(Might require moving this lemma to the end of the `contra` section after `tsub_tsub_cancel_of_le`.

Reply:
It does work, thanks! I changed it.